src/Service.Tests/OpenApiDocumentor/ParameterValidationTests.cs

```diff
@@ -0,0 +1,176 @@
+            }
+            else
+            {
+                // Get All method with path /entityName, will have no path parameters.
+                Assert.IsTrue(pathItem.Parameters.Count is 0);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Validates that Query parameters are correctly generated for GET methods in Table/Views.
+    /// $select, $filter, $orderby, $first, $after are the query parameters.
+    /// </summary>
+    /// <param name="entityName">The name of the entity.</param>
+    /// <param name="objectName">The name of the database object.</param>
+    /// <param name="entitySourceType">The source type of the entity.</param>
+    /// <returns></returns>
+    [DataTestMethod]
+    [DataRow("BooksTable", "books", EntitySourceType.Table, DisplayName = "Table with query parameters")]
+    [DataRow("BooksView", "books_view_all", EntitySourceType.View, DisplayName = "View with query parameters")]
+    public async Task ValidateQueryParametersForTablesAndViews(string entityName, string objectName, EntitySourceType entitySourceType)
+    {
+        EntitySource entitySource = new(Object: objectName, entitySourceType, null, null);
+        OpenApiDocument openApiDocument = await GenerateOpenApiDocumentForGivenEntity(entityName, entitySource);
+
+        foreach (OpenApiPathItem pathItem in openApiDocument.Paths.Values)
+        {
+            foreach ((OperationType operationType, OpenApiOperation operation) in pathItem.Operations)
+            {
+                if (operationType is OperationType.Get)
+                {
+                    Assert.IsTrue(operation.Parameters.Count is 5);
+
+                    // Assert that it contains all the query parameters with appropriate type
+                    Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name is "$select" && param.Schema.Type is "string"));
+                    Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name is "$filter" && param.Schema.Type is "string"));
+                    Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name is "$orderby" && param.Schema.Type is "string"));
+                    Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name is "$first" && param.Schema.Type is "integer"));
+                    Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name is "$after" && param.Schema.Type is "string"));
+                }
+                else
+                {
+                    // No query parameters for other OperationTypes. 
+                    Assert.IsFalse(operation.Parameters.Any(param => param.In is ParameterLocation.Query));
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Validates that Query parameters are correctly generated for Stored Procedures.
```

Comment:
```suggestion
    /// Validates that Query parameters are generated for Stored Procedures.
```

Are you validating more than the presence of the stored procedure query parameters? or are you performing additional validation on parameter metadata? if you are performing more than existence checks, can you elaborate here?